File: hackerank/CircleCity.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class CircleCity {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while(t>0)
        {
          int suburbs=0;
          String[] inp = sc.nextLine().split(" ");
          int rSquare = Integer.parseInt(inp[0]);
          int k = Integer.parseInt(inp[1]);
          //int r = (int)Math.sqrt(rSquare);
          for(int x=1;x*x<=rSquare;x++)
          {
            double y = Math.sqrt(rSquare - x*x);
            int yInt = (int)y;
            if(y-yInt<=0.00000001)
                suburbs+=4;
          }
          
          if(suburbs<=k)
              System.out.println("possible");
          else
              System.out.println("impossible");
          --t;
        }
    }
}

File: hackerank/circle_city.cpp
#include <iostream>
#include <cmath>

double epsilon = 0.000001;
using namespace std;

int main()
{
	int t;

	cin >> t;
	while(t--)
	{
		int r,k;
		cin >> r >> k;
		int count = 0;

		for(int y=0;y*y<r;y++)
		{
			double X = sqrt(r - y*y);
			int x = (int)X;
			if(abs(x-X) <= epsilon)
				count+=4;
		}
		if(count>k)
			cout<<"impossible"<<endl;
		else
			cout<<"possible"<<endl;
	}
}